src/nci_adapter.c

```diff
@@ -127,11 +138,151 @@ nci_adapter_intf_info_new(
         } else {
             info->activation_param.bytes = NULL;
         }
+
+        if(ntf->mode_param) {
+            info->mode_param_parsed = nci_util_copy_mode_param(ntf->mode_param,
+                ntf->mode);
+        } else {
+            info->mode_param_parsed = NULL;
+        }
         return info;
     }
     return NULL;
 }
 
+static
+inline
+gboolean
+mode_param_match_poll_a(
+    const NciModeParamPollA* prev,
+    const NciModeParamPollA* curr)
+{
+    /* compare all fields except UID 'cause UID may be
+    changed after losing field */
+    return prev->sel_res == curr->sel_res &&
+        prev->sel_res_len == curr->sel_res_len &&
+        !memcmp(prev->sens_res, curr->sens_res,
+            FIELD_SIZEOF(NciModeParamPollA, sens_res));
+}
+
+static
+inline
+gboolean
+mode_param_match_poll_b(
+    const NciModeParamPollB* prev,
+    const NciModeParamPollB* curr)
+{
+    /* compare all fields except UID 'cause UID may be
+    changed after losing field */
+    return prev->fsc == curr->fsc &&
+        !memcmp(prev->app_data, curr->app_data,
+            FIELD_SIZEOF(NciModeParamPollB, app_data)) &&
+        prev->prot_info.size == curr->prot_info.size &&
+        !memcmp(prev->prot_info.bytes, curr->prot_info.bytes,
+            curr->prot_info.size);
+}
+
+static
+inline
+gboolean
+mode_param_match_poll_a_t2(
+    const NciModeParamPollA* prev,
+    const NciModeParamPollA* curr)
+{
+    gboolean partial_match =  mode_param_match_poll_a(prev, curr);
+    /* For tag type 2 logic is almost the same, but random UID has some
+    limitations: according to AN10927 Random UID RID should be handled
+    separately - single sized (4 bytes) starting with 0x08 */
+    if(prev->nfcid1_len == curr->nfcid1_len &&
+        curr->nfcid1_len == RANDOM_UID_SIZE &&
+        prev->nfcid1[0] == curr->nfcid1[0] &&
+        curr->nfcid1[0] == RANDOM_UID_START_BYTE) {
+        return partial_match;
+    } else {
+    /* otherwise UID should fully match */
```

Comment:
Align the comment with the next line, **C**apitalize the first word for consistency